editor/src/components/canvas/controls/grid-controls.spec.browser2.tsx

```diff
@@ -0,0 +1,343 @@
+      width: 50,
+      height: 50,
+    })
+    expect(result).toEqual({
+      right: { x: 777.5, y: 315.5, width: 12, height: 1 },
+      top: { x: 751.5, y: 285.5, width: 1, height: 5 },
+    })
+  })
+  it('pinned bottom and left, grid rows and columns fully specified and not absolutely positioned', async () => {
+    const result = await testGridOutlines({
+      bottom: 5,
+      left: 12,
+      gridRowStart: 2,
+      gridColumnStart: 3,
+      gridRowEnd: 2,
+      gridColumnEnd: 3,
+      width: 50,
+      height: 50,
+    })
+    expect(result).toEqual({})
+  })
+  it('pinned bottom and left, grid rows and columns fully specified and absolutely positioned', async () => {
+    const result = await testGridOutlines({
+      position: 'absolute',
+      bottom: 5,
+      left: 12,
+      gridRowStart: 2,
+      gridColumnStart: 3,
+      gridRowEnd: 2,
+      gridColumnEnd: 3,
+      width: 50,
+      height: 50,
+    })
+    expect(result).toEqual({
+      left: { x: 688.5, y: 355.5, width: 12, height: 1 },
+      bottom: { x: 725.5, y: 381.5, width: 1, height: 5 },
+    })
+  })
+  it('pinned bottom and right, grid rows and columns fully specified and not absolutely positioned', async () => {
+    const result = await testGridOutlines({
+      bottom: 5,
+      right: 12,
+      gridRowStart: 2,
+      gridColumnStart: 3,
+      gridRowEnd: 2,
+      gridColumnEnd: 3,
+      width: 50,
+      height: 50,
+    })
+    expect(result).toEqual({})
+  })
+  it('pinned bottom and right, grid rows and columns fully specified and absolutely positioned', async () => {
+    const result = await testGridOutlines({
+      position: 'absolute',
+      bottom: 5,
+      right: 12,
+      gridRowStart: 2,
+      gridColumnStart: 3,
+      gridRowEnd: 2,
+      gridColumnEnd: 3,
+      width: 50,
+      height: 50,
+    })
+    expect(result).toEqual({
+      right: { x: 777.5, y: 355.5, width: 12, height: 1 },
+      bottom: { x: 751.5, y: 381.5, width: 1, height: 5 },
+    })
+  })
+
+  it('pinned top and left, grid column end set to auto and not absolutely positioned', async () => {
+    const result = await testGridOutlines({
+      top: 5,
+      left: 12,
+      gridRowStart: 2,
+      gridColumnStart: 3,
+      gridRowEnd: 2,
+      gridColumnEnd: 'auto',
+      width: 50,
+      height: 50,
+    })
+    expect(result).toEqual({})
+  })
+  it('pinned top and left, grid column end set to auto and absolutely positioned', async () => {
+    const result = await testGridOutlines({
+      position: 'absolute',
+      top: 5,
+      left: 12,
+      gridRowStart: 2,
+      gridColumnStart: 3,
+      gridRowEnd: 2,
+      gridColumnEnd: 'auto',
+      width: 50,
+      height: 50,
+    })
+    expect(result).toEqual({
+      left: { x: 688.5, y: 315.5, width: 12, height: 1 },
+      top: { x: 725.5, y: 285.5, width: 1, height: 5 },
+    })
+  })
+  it('pinned top and right, grid column end set to auto and not absolutely positioned', async () => {
```

Comment:
for the "...end set to auto" tests, we should add a padding to the grid parent – because `auto` corrseponds to the padding edge, which is only different from the last line if there is a padding set